otel/otel.go

```diff
@@ -0,0 +1,171 @@
+package otel
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	otelsdk "go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/metric"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+)
+
+// Config defines the Otel Wrapper configuration for metrics
+type Config struct {
+	ServiceName               string
+	Debug                     bool
+	ExporterType              string
+	ExporterEndpoint          string
+	ExporterTimeout           string
+	ExporterCertificate       string
+	ExporterClientCertificate string
+	ExporterClientKey         string
+	CollectionInterval        time.Duration
+	MetricsReader             sdkmetric.Reader
+	MetricsExporter           sdkmetric.Exporter
```

Comment:
I also think that these information should not be keeped in Config struct. But in the struct that will be keeped globally through the  `singleton`